File: 004.py
#!/usr/bin/env python

"""Find the largest palindrome made from the product of two 3-digit numbers."""

def is_palindrome(n):
    """Determine if n is a palindrome."""
    for i, c in enumerate(n[:len(n)/2]):
        if c != n[(len(n)-1) - i]:
            return False
    return True

max = 0
for a in range(999, 100, -1):
    for b in range(999, 100, -1):
        if (a * b > max) and is_palindrome(str(a * b)):
            max = a * b
print max

File: 004.swift
#!/usr/bin/xcrun swift

// Find the largest palindrome made from the product of two 3-digit numbers.

func isPalindrome(token: String) -> Bool {
    // Determine if token is a palindrome.
    let characters = Array(token)
    for (i, c) in enumerate(characters[0..<characters.count / 2]) {
        if c != characters[(characters.count - 1) - i] {
            return false
        }
    }
    return true
}

var max = 0
for var a = 999; a >= 100; --a {
    for var b = 999; b >= 100; --b {
        if (a * b > max) && isPalindrome(String(a * b)) {
            max = a * b
        }
    }
}
println(max)
